### DISCUSSION TITLE
Device group operations without 'du' service

### DISCUSSION THREAD

**Discussion by ranjith-t** [Azure/azure-iot-sdk-java#1770] [Q&A] [answered]
Hi,
  I was under the impression that the device groups are based on tags and tag values in the IoTHub. however, from the documentation it looks like they are tied to 'device update' service. For example a device group delete CLI listed 'az iot du device group delete' is with in the 'du' service.  
  what is the purpose of 'du' if we are not doing deployment but only operations such as direct methods and twin update?
  
  does this mean that the SDK service will not have a API for device group?  Can someone explain how a device group can be created, deleted without device update service? 
  As per me, by adding tags to list of devices they will form a device group. And when the tags are removed from all the devices, the device group is removed/deleted. For any device group operation, the filter would be the tag/value to simulate a device group. Am I missing something ?

**Comment by timtay-microsoft** [comment]
This may be a better question in [this repo](https://github.com/Azure/azure-sdk-for-java/) since that is where the [Device Update SDK](https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/deviceupdate/azure-iot-deviceupdate) lives and people like @dpokluda (who manages that SDK) can provide more context for you.

As for this SDK, we don't plan on adding any APIs for Device Update operations since those are not part of the [Azure IoT Hub API surface](https://learn.microsoft.com/en-us/rest/api/iothub/service/operation-groups?view=rest-iothub-service-2020-03-13). As far as I'm aware, you will need to use the Azure Device Update SDK to do operations like the ones you are describing.

**Reply by ranjith-t** [reply]
ok, I understand that device updates are part of the other SDK. My question is more about APIs such as scheduleUpdateTwin() nad scheduleDirectMethod() in scheduled job clients. Isn't these suppose to support device group as well? what is the reasoning for using device update service for these functionalities when there is no update is happening? Isn't this a overhead for the system?

also, if these functionalities are moved to device update SDK? if so, does that mean these APIs are obsolete?

**Reply by timtay-microsoft** [reply, marked as answer]
> Isn't these suppose to support device group as well? what is the reasoning for using device update service for these functionalities when there is no update is happening?

I don't have enough context on the device update APIs to help you on that. Someone on the other thread you opened should be able to help here, though.

> also, if these functionalities are moved to device update SDK? if so, does that mean these APIs are obsolete?

No, this SDK is still the recommended way to interact with an IoT hub (other than doing device update operations). The Device Update SDK and service are both complements to solutions that use IoT hub. They don't replace any feature of IoT hub.
